source/particle/property/viscoplastic_strain_invariants.cc

```diff
@@ -50,6 +51,8 @@ namespace aspect
 
         n_components = 0;
         material_inputs = MaterialModel::MaterialModelInputs<dim>(1,this->n_compositional_fields());
+        material_outputs = MaterialModel::MaterialModelOutputs<dim>(1, this->n_compositional_fields());
+
 
         // Find out which fields are used.
         if (this->introspection().compositional_name_exists("plastic_strain"))
```

Comment:
Replace the part below with this to set the compositional field indices:
```suggestion
        if (this->introspection().compositional_name_exists("plastic_strain"))
          {
              n_components += 1;
            plastic_strain_index = this->introspection().compositional_index_for_name("plastic_strain");
          }
          
                  if (this->introspection().compositional_name_exists("viscous_strain"))
          {
              n_components += 1;
            plastic_strain_index = this->introspection().compositional_index_for_name("viscous_strain");
          }
          
                  if (this->introspection().compositional_name_exists("total_strain"))
          {
              n_components += 1;
            total_strain_index = this->introspection().compositional_index_for_name("plastic_strain");
          }
```